sql/kvstore/commitment.go

```diff
@@ -0,0 +1,32 @@
+package kvstore
+
+import (
+	"fmt"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/sql"
+)
+
+const commitmentATXKey = "commitmentATX"
+
+func getKeyForNode(nodeId types.NodeID) string {
+	return fmt.Sprintf("%s-%s", commitmentATXKey, nodeId)
+}
+
+// AddCommitmentATXForNode adds the id for the commitment atx to the key-value store.
+func AddCommitmentATXForNode(db sql.Executor, atx types.ATXID, nodeId types.NodeID) error {
+	return addKeyValue(db, getKeyForNode(nodeId), &atx)
+}
+
+// GetCommitmentATXForNode returns the id for the commitment atx from the key-value store.
+func GetCommitmentATXForNode(db sql.Executor, nodeId types.NodeID) (types.ATXID, error) {
+	var res types.ATXID
+	if err := getKeyValue(db, getKeyForNode(nodeId), &res); err != nil {
+		return *types.EmptyATXID, err
+	}
+	return res, nil
+}
+
+func ClearCommitmentAtx(db sql.Executor, nodeId types.NodeID) error {
```

Comment:
why is this necessary? isn't it true that once stored for a node ID it should never change?
in another words, it should incur an error if the code tries to overwrite a kv pair

Reply:
Yes and no, after **publishing** the initial ATX the commitment ATX never changes for a node. Before that a node is free to choose any commitment ATX it deems satisfactory. After the initial ATX has been published we don't need the kvstore any more to get the commitment ATX, we can just get it from the first ATX published by the node.

The kvstore is not a source of truth for the commitment ATX, it is only a transient storage for a node that initialises PoST and hasn't published its initial ATX yet.

Reply:
agreed. but current code does not try to choose a better committment ATX before it publish the initial ATX.
having this interface is confusing until code is added to choose a better one.

Reply:
OK, then I'll remove the `ClearCommitmentAtx` function and update the setter and getters for commitmentATX